### PR TITLE
Add Unmount component

### PR DESCRIPTION
**Pull request by Iamsamwhoami**

### CODE REVIEW

File: src/unmount.test.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+test("should call on function when mounted", () => {
+  const mockOn = jest.fn<Function>(jest.fn());
+
+  const wrapper = mount(
+    <Unmount on={mockOn}>
+      <div />
+    </Unmount>
+  );
+
```
Comment (palmfjord) [inline src/unmount.test.tsx]: Maybe we should expect it hasn't been called here?

Comment (Iamsamwhoami) [reply src/unmount.test.tsx]: @palmfjord ah, done!

File: src/unmount.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+
+interface UnmountProps {
+  on: Function;
```
Comment (palmfjord) [inline src/unmount.tsx]: imo something like `on: (props: UnmountProps) => void` would be better, so we actually get type checking on the function calls too. `Function` is from my experience a bit like `any`, and `any` i evil imo

Comment (Iamsamwhoami) [reply src/unmount.tsx]: @palmfjord there is actually zero benefit in passing props to on, so removing that and changing type annotation to `() => void`.

**Comment by Iamsamwhoami** [comment]
I merged it 🙈hate more if you wanna and I'll PR again 😄

**Comment by palmfjord** [comment]
@Iamsamwhoami hehe, sneaky ;) i would approve if i could now ✅

**Comment by Iamsamwhoami** [comment]
@palmfjord work fast https://media.giphy.com/media/XIqCQx02E1U9W/giphy.gif

**Pull request merged**